letta/credentials.py

```diff
@@ -30,14 +30,14 @@ class LettaCredentials:
 
     # azure config
     azure_auth_type: str = "api_key"
-    azure_key: Optional[str] = None
+    azure_key: Optional[str] = os.getenv("AZURE_OPENAI_API_KEY")
 
     # groq config
     groq_key: Optional[str] = os.getenv("GROQ_API_KEY")
 
     # base llm / model
-    azure_version: Optional[str] = None
-    azure_endpoint: Optional[str] = None
+    azure_version: Optional[str] = "2023-03-15-preview"  # None
```

Comment:
you probably want to remove this defaults, since it can make debugging confusing

Reply:
SORRY i should've caught this, this is debugging litter from when i started looking at azure last week :\